File: src/main/java/link/pihda/billofmaterial/service/ProcurementService.java
/*
 * Copyright (c) 2023.  Adib S. A.
 * Copyright (c) 2023.  PT Atadopos
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 */

package link.pihda.billofmaterial.service;

import link.pihda.billofmaterial.dao.ProcurementDao;
import link.pihda.billofmaterial.entity.Procurement;
import link.pihda.billofmaterial.entity.Transaction;

import java.util.List;

public class ProcurementService {
    private final ProcurementDao procurementDao = new ProcurementDao();

    public void save(Procurement procurement) {
        procurementDao.save(procurement);
    }

    public List<Procurement> getAll() {
        return procurementDao.getAll();
    }

    public void delete(Procurement procurement) {
        procurementDao.delete(procurement);
    }

    public void update(Procurement procurement) {
        procurement.setTotalPrice(procurement.getTransactions().stream().mapToDouble(Transaction::getTotalPrice).sum());
        procurementDao.update(procurement);
    }
}

File: src/main/kotlin/link/pihda/billofmaterial/service/ProcurementService.kt
/*
 * Copyright (c) 2023.  Adib S. A.
 * Copyright (c) 2023.  PT Atadopos
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 */
package link.pihda.billofmaterial.service

import link.pihda.billofmaterial.dao.ProcurementDao
import link.pihda.billofmaterial.entity.Procurement

class ProcurementService {
    private val procurementDao = ProcurementDao()

    fun save(procurement: Procurement) {
        procurementDao.save(procurement)
    }

    fun getAll(): List<Procurement> {
        return procurementDao.getAll()
    }

    fun delete(procurement: Procurement) {
        procurementDao.delete(procurement)
    }

    fun update(procurement: Procurement) {
        procurement.totalPrice = procurement.transactions.sumOf { it.totalPrice }
        procurementDao.update(procurement)
    }
}
